README.md

```diff
@@ -6,39 +6,184 @@ Please edit this README.md file with your answers to the worksheet questions.
+|false|false|true |false|
+|false|true |false|true |
+|false|true |true |true |
+|true |false|false|true |
+|true |false|true |false|
+|true |true |false|true |
+|true |true |true |true |
+
+
 ### d
 
+|A   |B   |C   |D   |A AND NOT (B OR NOT C) AND (NOT A AND D)   |
+|-----|-----|-----|-----|-----|
+|false|false|false|false|false|
+|false|false|false|true |false|
+|false|false|true |false|false|
+|false|false|true |true |false|
+|false|true |false|false|false|
+|false|true |false|true |false|
+|false|true |true |false|false|
+|false|true |true |true |false|
+|true |false|false|false|false|
+|true |false|false|true |false|
+|true |false|true |false|false|
+|true |false|true |true |false|
+|true |true |false|false|false|
+|true |true |false|true |false|
+|true |true |true |false|false|
+|true |true |true |true |false|
+
 ## Question 2
 
 ### a
 
+![Alt Text](https://i.imgur.com/WFUVxDa.png)
+
 ### b
 
+![Alt Text](https://i.imgur.com/6zLcBpK.png)
+
 ### c
 
+![Alt Text](https://i.imgur.com/5e9qIyC.png)
+
 ### d
 
+![Alt Text](https://i.imgur.com/2sTsM3m.png)
+
 ## Question 3
 
 ### a
 
+|A   |B   |NOT (A OR B)   |
+|-----|-----|-----|
+|false|false|true|
+|false|true|false |
+|true|false |false|
+|true|true |false |
+
+|A   |B   |NOT A AND NOT B   |
+|-----|-----|-----|
+|false|false|true|
+|false|true|false |
+|true|false |false|
+|true|true |false |
+
 ### b
 
+|A   |B   |NOT (A AND B)   |
+|-----|-----|-----|
+|false|false|true|
+|false|true|true |
+|true|false |true|
+|true|true |false |
+
+|A   |B   |NOT A OR NOT B   |
+|-----|-----|-----|
+|false|false|true|
+|false|true|true |
+|true|false |true|
+|true|true |false |
+
 ### c
 
+|A    |B    |C    |(A AND B) OR (A AND C)|
+|-----|-----|-----|-----|
+|false|false|false|false|
+|false|false|true |false|
+|false|true |false|false|
+|false|true |true |false|
+|true |false|false|false|
+|true |false|true |true|
+|true |true |false|true |
+|true |true |true |true |
+
+|A    |B    |C    |	A AND (B OR C)|
+|-----|-----|-----|-----|
+|false|false|false|false|
+|false|false|true |false|
+|false|true |false|false|
+|false|true |true |false|
+|true |false|false|false|
+|true |false|true |true|
+|true |true |false|true |
+|true |true |true |true |
+
 ### d
 
+
+|A   |B   |C  |(A OR B) AND (A OR C)|
+|-----|-----|-----|-----|
+|false|false|false|false|
+|false|false|true|false |
+|false|true|false |false|
+|false|true|true |true |
+|true|false |false|false|
+|true|false|true|true |
+|true|true |false |false|
+|true|true |true |true |
+
+|A   |B   |C  |(A OR (B AND C)|
+|-----|-----|-----|-----|
+|false|false|false|false|
+|false|false|true|false |
+|false|true|false |false|
+|false|true|true |true |
+|true|false |false|false|
+|true|false|true|true |
+|true|true |false |false|
+|true|true |true |true |
+
 ## Question 4
 
 ### a
 
+NOT (A OR B) = NOT A AND NOT B
+
 ### b
 
+(A AND B) OR (A AND C) = A AND (B OR C)
+
 ### c
 
+NOT (A AND B) = NOT A OR NOT B
+
 ### d
 
+(A OR B) AND (A OR C) = A OR (B AND C)
```

Comment:
Right idea, but needs a bit more explanation as to how these correspond to the programs